README.md

```diff
@@ -24,52 +24,86 @@ The 2048 Game project is a Python implementation of the popular 2048 puzzle game
 - **Random Number Generator**: Spawn new tiles with random values.
 - **Past Grid Tracking**: Keep track of the previous grid states using a stack.
 - **Score Tracking**: Each move the score get updated
-- **Game Over and Victory Detection**: Detect when the game is over or when the player wins.
+- **Undo Function**: Allows players to revert to the previous grid state, facilitating strategic moves and error correction.
+- **Game Over and Victory Detection**: Detect when the game is over or when the player wins. And also **Quit** possibility
+- **Ranking System**: Maintaining a ranking of 10 most high scores.
+- **Tree-Based Move Suggestions**: Provides approximate suggestions for optimal moves using a tree structure.
 
 ## Data Structures Used
 
 - **2D Array**: Representing the game grid.
 - **Stack** (linked list-based): Storing past grid states.
 - **Dictionary**: Managing key game features.
 - **Set**: Use a set to keep track of empty celleces (tiles with value 0)
+- **Tree**: Used in the tree-based move suggestion system.
 
 ## Algorithms
 
 - **Merge Algorithm**: Sum two cells adjacent in the same direction of movement.
 - **Random Number Generator Algorithm**: Generate random numbers in the grid.
 - **Move Algorithm**: Move the grid in the specified direction.
+- **Sorting Algorithms (QuickSort)**: Used in ranking system to sort the high scores.
 
 ## Code Structure Explanation
 
-The code is organized into several components to enhance readability and maintainability.
+The code is organized into several components to enhance readability and maintainability:
+
+- **`main.py`**: The entry point of the game. It initializes the `GameBoard` and runs the main game loop.
+
+- **`src` folder**: Contains various modules and classes:
+  
+  - _game_class.py_: Contains the GameBoard class with methods for game mechanics.
+
+  - _stack.py_: Implements the Stack class for undo functionality.
+
+  - _tree_class.py_: Defines the TreePossibilities class for move suggestions.
+
+  - _tree.py:_ Implements a tree with an array of children
+
+
 
 ### `GameBoard` Class
 
 The `GameBoard` class is the core of the game logic and user interaction. It contains the following methods:
 
```

Comment:
metion also 
- self.pastgrids = Stack() # initialize a stack to keep track of past grids
- self.score = 0  # initialize user score to 0